_includes/patterns-filter.html

```diff
@@ -403,32 +413,41 @@
 
   // Function called when user clicks on any of the Type radio buttons
   function filterByType(value){
-
-    // Deselect all the checkboxes when radio button is selected
-    document.querySelectorAll("input[type=checkbox]:checked").forEach(function (el) {
-      el.checked = false;
-    });
-
-    // Loop over all patern cards and check if the "type" of pattern card matches with the radio button value received as argument to function
-    // If yes, show the card, else hide it
-    // Show all cards if "All" radio button is selected
-    const patternCards = document.querySelectorAll(".patternCard");
-    for (let j = 0; j < patternCards.length; j++) {
-      const filter = patternCards[j].getAttribute("type").toLowerCase();
-      let showCard = false;
-      if (value === "All" || filter.includes(value)) {
-        showCard = true;
+    // if no technology is selected, and the category is set to "all" then render all the cards
+      if(value.toLowerCase() == "all" && globalTechnologyList.length == 0){
+        globalCategorySelected = "All";
+        for (let j = 0; j < patternCards.length; j++) {
+          patternCards[j].style.display = "block";
+        }
+        return;
       }
-      if (showCard) {
-        patternCards[j].style.display = "block";
-      } else {
-        patternCards[j].style.display = "none";
+      // if no technology is selected but we have a category value,
+      // Loop over all patern cards and check if the "type" of pattern card matches with the radio button value received as argument to function
+      // If yes, show the card, else hide it
+      if(globalTechnologyList.length == 0){
+          globalCategorySelected = value.toLowerCase();
+          for (let j = 0; j < patternCards.length; j++) {
+            const filter = patternCards[j].getAttribute("type").toLowerCase();
+            let showCard = false;
+            if (filter.includes(value.toLowerCase())) {
+              showCard = true;
+            }
+            if (showCard) {
+              patternCards[j].style.display = "block";
+            } else {
+              patternCards[j].style.display = "none";
+            }
```

Comment:
This if else condition seems to be using multiple time, maybe we can extract it to a function